backend/app/api/endpoints/repository.py

```
@@ -3,35 +3,49 @@
+    repository_service,
+    snapshot_repository_user,
+)
 
 # Logger instance
 logger = logging.getLogger(__name__)
 
 router = APIRouter()
 
 
+def get_repository_user_context(
+    current_user: User = Depends(security.get_current_user),
```

Ruff (0.15.15)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)